### ISSUE TITLE
Stamina

### ISSUE THREAD

**Issue by IntegratedQuantum** [PixelGuys/Cubyz#257]
Stamina is replenished naturally, consuming hunger.
Not all actions consume stamina. For example we don't want to slow down exploration, so sprinting does not consume stamina, but you can only sprint if your stamina is above a certain level.
When using a tool or weapon stamina decreases depending on its weight.

This has the potential to add a lot of depth to fighting and crafting. For example a low-range dagger might mean that you get hit more often, but consumes less stamina than a long range spear.
But as you get more stamina conserving in your combat style, or against certain bosses, the spear might be the better option.

**Comment by archbirdplus** [comment]
Stamina has multiple aspects to it. Of course, there is the global "running out of air" state. But each muscle has its own temporary store of oxygen/energy which it can use for some amount of time. That means that each muscle could be tracked independently. There's also a several categories of muscle fibers, which each use an/aerobic respiration and are used at different times.
This adds even more complexity to the game. Anaerobic respiration, for example, probably depletes hunger more quickly due to being less efficient. This means that running and sprinting could have slightly different stamina trackers. It also means that in one Cubyz day, it could make sense to split up work by muscle group. (Aside: sleep quality aiding stamina recovery could encourage players to build temporary shelters at night while exploring to allow for more optimal performance the next day)
On the topic of exercise, even more complexity could be made from simulating muscle growth. Cubyz could even consider the quality of sleep, or the amount of carbohydrates, fats, and amino acids that the player consumes throughout the day, compare that with the types of exercise each muscle had performed...

**Comment by IntegratedQuantum** [comment]
Let's not overcomplicate things. A single "out of air"-stamina value is a good enough approximation that is easy to communicate to the player.

About sleep: it is an interesting idea to give the player a stat boost for sleeping well (though that is a different topic #258).
About exercise: That would basically be a skilling/levelling system. I'm not yet sure if I want something like that. Skilling often makes players grind more, and I don't know if I want that.
About nutrients: While it would be great to encourage eating different foods, it would put some pressure on biome design, since all nutrients should be accessible in most surface biomes. So I would like to leave it open for now.

**Comment by IntegratedQuantum** [comment]
I don't think this is a strong enough use case to justify adding another bar to the UI.
Furthermore I think we might be able to achieve similar mechanical depth with the energy bar concept.
